book_source/06_reference/04_docker/03_architecture.Rmd

```diff
@@ -17,5 +19,186 @@ The PEcAn framework containers consist of multiple unique ways to interact with
+	
+	- `pecan` -- This volume contains PEcAn's `dbfiles`, which include downloaded and converted model inputs, processed configuration files, and outputs.
+	It is used by almost all of the services in the PEcAn stack, and is typically mounted to `/data`.
+ 
+#### `traefik` {#pecan-dc-traefik}
+
+[Traefik](https://traefik.io/) manages communication among the different PEcAn services and between PEcAn and the web.
+Among other things, `traefik` facilitates the setup of web access to each PEcAn service via common and easy-to-remember URLs.
+For instance, the following lines in the `web` service configure access to the PEcAn web interface via the URL http://localhost:8000/pecan/ :
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services$web["labels"], stdout())
+```
+
+(Further details in the works...)
+
+The traefik service configuration looks like this:
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services["traefik"], stdout())
+```
+
+#### `minio` {#pecan-dc-minio}
+
+[Minio](https://github.com/minio/minio) is a browser-based file manager that facilitates exploration of PEcAn files without the need for CLI tools.
+
+Our current configuration is as follows:
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services["minio"], stdout())
+```
+
+(Further details in the works...)
+
+#### `postgres` {#pecan-dc-postgres}
+
+This service provides a working PostGIS database.
+Our configuration is fairly straightforward:
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services["postgres"], stdout())
+```
+
+Some additional details about our configuration:
+
+- `image` -- This pulls a container with PostgreSQL + PostGIS pre-installed.
+Note that by default, we use PostgreSQL version 9.5.
+To experiment with other versions, you can change `9.5` accordingly.
+
+- `networks` -- This allows PostgreSQL to communicate with other containers on the `pecan` network.
+As mentioned above, the hostname of this service is just its name, i.e. `postgres`, so to connect to the database from inside a running container, use a command like the following: `psql -d bety -U bety -h postgres`
+
+- `volumes` -- Note that the PostgreSQL data files (which store the values in the SQL database) are stored on a _volume_ called `postgres` (which is _not_ the same as the `postgres` _service_, even though they share the same name).
+
+#### `rabbitmq` {#pecan-dc-rabbitmq}
+
+[RabbitMQ](https://www.rabbitmq.com/) is a message broker service.
+In PEcAn, RabbitMQ functions as a task manager and scheduler, coordinating the execution of different tasks (such as running models and analyzing results) associated with the PEcAn workflow.
+
+Our configuration is as follows:
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services["rabbitmq"], stdout())
+```
+
+Note that the `traefik.frontend.rule` indicates that browsing to http://localhost:8000 (with no further tags) leads to the RabbitMQ management console.
```

Comment:
Since the default username and password is used (guest/guest) for rabbitmq we highly recommend changing this for the production instances of PEcAn, as well as removing access to the rabbitmq interface using traeffic.

Reply:
Addressed in 8745aeb9a.